include/rocksdb/comparator.h

```diff
@@ -33,6 +33,35 @@ class CompareInterface {
   virtual int Compare(const Slice& a, const Slice& b) const = 0;
 };
 
+// The general interface for comparing two Slices that are related to
+// timestamps.
+// They are defined for both Comparator and some internal data structures.
+class CompareWithTimestampInterface {
```

Comment:
Would we ever have a Compare that was a CompareWithTimestamp that is not also a CompareInterface?  If not, is there a reason why to not have CompareWithTimestamp extend CompareInterface?

Reply:
CompareInterface is used for Internal comparators, and internal operators don't seem to do those timestamp comparisons.